ui/selectors/selectors.js

```diff
@@ -1278,13 +1279,59 @@ export function getAllAccountsOnNetworkAreEmpty(state) {
   return hasNoNativeFundsOnAnyAccounts && hasNoTokens;
 }
 
-export function getShouldShowSeedPhraseReminder(state) {
-  const { tokens, seedPhraseBackedUp, dismissSeedBackUpReminder } =
-    state.metamask;
+export function getIsAnyAccountWithBalanceOnNonTestNetwork(state) {
+  const { cachedBalances } = state.metamask;
+
+  const zeroBalanceCheck = (currentValue) =>
+    currentValue === '0x0' || currentValue === '0x00';
+
+  const balancesOnNonTestNetworks = Object.entries(cachedBalances)
+    .filter(
+      (networkAndAccountBalances) =>
+        TEST_CHAINS.includes(networkAndAccountBalances[0]) === false &&
+        networkAndAccountBalances,
+    )
+    .map((accountAndBalance) => Object.values(accountAndBalance[1]))
+    .map((balance) => balance.every(zeroBalanceCheck));
```

Comment:
Rather than looping over `Object.entries` and its results 3 times, would it be possible to use `reduce` to get the desired outcome?

Reply:
I tried, but didn't manage to get the proper result I needed...